drivers/storage/portworx/deployment_test.go

```diff
@@ -760,54 +776,41 @@ func TestPodSpecWithCloudStorageSpec(t *testing.T) {
 		MaxStorageNodesPerZone: &userProvidedInstancesPerZoneUint32,
 	}
 
-	mockStorageManager.EXPECT().
-		GetStorageDistribution(&cloudops.StorageDistributionRequest{
-			ZoneCount:        len(zoneToInstancesMap),
-			InstancesPerZone: userProvidedInstancesPerZone,
-			UserStorageSpec: []*cloudops.StorageSpec{
-				{
-					IOPS:        uint32(100),
-					MinCapacity: uint64(100),
-					MaxCapacity: uint64(200),
-				},
-				{
-					IOPS:        uint32(200),
-					MinCapacity: uint64(200),
-					MaxCapacity: uint64(500),
-				},
-			},
-		}).
-		Return(&cloudops.StorageDistributionResponse{
-			InstanceStorage: []*cloudops.StoragePoolSpec{
-				{
-					DriveCapacityGiB: uint64(120),
-					DriveType:        "foo",
-					DriveCount:       1,
-					InstancesPerZone: userProvidedInstancesPerZone,
-					IOPS:             uint32(110),
-				},
-				{
-					DriveCapacityGiB: uint64(220),
-					DriveType:        "bar",
-					DriveCount:       1,
-					InstancesPerZone: userProvidedInstancesPerZone,
-					IOPS:             uint32(210),
-				},
-			},
-		}, nil)
+	expectedCloudStorageSpec := []corev1alpha1.StorageNodeCloudDriveConfig{
+		{
+			Type:      "foo",
+			SizeInGiB: uint64(120),
+			IOPS:      uint32(110),
+			Options:   map[string]string{"foo1": "bar1"},
+		},
+		{
+			Type:      "bar",
+			SizeInGiB: uint64(220),
+			IOPS:      uint32(210),
+			Options: map[string]string{
+				"foo3": "bar3",
+			}},
+	}
 
 	expectedArgs = []string{
 		"-c", "px-cluster",
 		"-x", "kubernetes",
 		"-s", "type=foo,size=120,iops=110,foo1=bar1",
 		"-s", "type=bar,size=220,iops=210,foo3=bar3",
-		"-max_storage_nodes_per_zone", "3",
+		"-max_storage_nodes_per_zone", "2",
 	}
 
-	actual, err = driver.GetStoragePodSpec(cluster)
+	// despite adding a new node, node spec should be pulled from the existing storage node for nodeName
+	// Storage node will be created for "nodeName1"
+	actual, err = driver.GetStoragePodSpec(cluster, nodeName)
 	assert.NoError(t, err, "Unexpected error on GetStoragePodSpec")
 	assert.ElementsMatch(t, expectedArgs, actual.Containers[0].Args)
 
+	list, err = driver.storageNodesList(cluster)
+	assert.NoError(t, err, "Unexpected error on GetStorageNodeList")
+	assert.Equal(t, 1, len(list), "expected storage nodes in list")
+	assert.ElementsMatch(t, list[0].Spec.CloudStorage.DriveConfigs, expectedCloudStorageSpec)
```

Comment:
This should be checked in the first case we well on line 756.

Reply:
right